src/core/identity.cc

```diff
@@ -50,19 +50,26 @@
 namespace i2p {
 namespace data {
 
+// TODO(unassigned): identity implementation needs a big refactor
+
+// TODO(unassigned): keep an eye open for alignment issues and for hacks like:
+// copy public and signing keys together
+//memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+
 Identity& Identity::operator=(const Keys& keys) {
-  // copy public and signing keys together
-  memcpy(public_key, keys.public_key, sizeof(public_key) + sizeof(signing_key));
+  memcpy(public_key, keys.public_key, sizeof(public_key));
```

Comment:
>What if we do want to copy the private key?

It should be done elsewhere. This is an abhorrent implementation.

>Does that ever happen?

Because we currently don't subscribe in-net, address book won't touch this code so, **currently** 100% no. Once `client_context.h:75` is resolved, then **always** 100% no.

After the TODO is resolved, we should also be able to begin a bigger refactor. We should also consider eliminating DSA-SHA1 entirely because **we can**.